client/landing/stepper/declarative-flow/with-theme-assembler-flow.ts

```diff
@@ -27,25 +27,29 @@ const withThemeAssemblerFlow: Flow = {
 		);
 		const { setSelectedDesign, setIntent } = useDispatch( ONBOARD_STORE );
 		const selectedTheme = useQuery().get( 'theme' );
+		const theme = useSelector( ( state ) => getTheme( state, 'wpcom', selectedTheme ) );
 
 		// We have to query theme for the Jetpack site.
 		useQueryTheme( 'wpcom', selectedTheme );
 
 		useEffect( () => {
-			if ( selectedTheme === BLANK_CANVAS_DESIGN.slug ) {
-				// User has selected blank-canvas-3 theme from theme showcase and enter assembler flow
-				setSelectedDesign( {
-					...selectedDesign,
-					...BLANK_CANVAS_DESIGN,
-					recipe: {
-						...selectedDesign?.recipe,
-						...BLANK_CANVAS_DESIGN.recipe,
-					},
-				} as Design );
+			if ( ! theme ) {
```

Comment:
Could we add some kind of logging here? Since seems that the page will silently fail if the theme cannot be resolved.